### DISCUSSION TITLE
2023/10/30: Inauguration 2.0

### DISCUSSION THREAD

**Discussion by tecimovic** [project-chip/zap#1188] [Steering Committee]
- Restart the steering committee meeting.
- Present: @bzbarsky-apple , @andreilitvin , @cecille , @tecimovic , @brdandu , @wehale , @paulr34 , @mkardous-silabs, @EtienneCampeauSilabs
- Discussion topics:
  - Move to CSA XML: integrate clusters one-by-one: Andrei will tell us which cluster to use. We start with a mixed state loading an individual cluster.
  - Feature Map: important. Should be done. Understand the spec well: what can override what, etc.
  - Externalized helpers: it would be helpful if there is at least SOME ability to write custom helper code on SDK. Find a middle ground between "full safety" and "do whatever you want".
  - Diffs of zap files are hard to read: don't duplicate XML data inside .zap file, will eliminate a lot of these diffs
  - Deal with what "Defaults" mean. Sort this out in ZAP.
  - add hover messages to buttons and such
  - deal with .zap/ deletion.
  - what does "Generate" buttons do.
  - zap UI assumes zigbee, not matter. There are things that make no sense to matter (such as "reportable" toggle)
  - Spec Check: less important than the above 2 for the Matter SDK developers.

- Notes:
 - zap is using "default" wrong. "Defaults" in the spec are the "default values" that the "receiver" should assume the values are, if they are omitted. They are not "default" values for attributes at startup, as Zap assumes.

**Comment by brdandu** [comment]
All issues related to this discussion can be found under events on the right. Feature map issue which is not under events can be tracked in https://github.com/project-chip/zap/issues/547 .

Couple of follow up questions on the discussion:
- There was a mention of the conformance check script which is already being used. what does this script take as input? Could we get a reference to that script?
- I heard attribute reporting is not used in Matter. Is the generated min max array in helper-endpointconfig not used as well in the Matter implementation? Do we need changes to generation as well?

**Reply by bzbarsky-apple** [reply]
> I heard attribute reporting is not used in Matter

That depends on how you define "attribute reporting".  It's hard to say what exactly you heard and how it maps to reality.

Matter has reporting of attribute changes.  It works quite differently from Zigbee's in many ways.

> Is the generated min max array in helper-endpointconfig not used as well in the Matter implementation?

It's used, for writable attributes.

**Comment by paulr34** [comment]
Via CSA slack: 

adding to the list of items; @bzbarsky-apple proposes when the user opens a new configuration then root node is configured on endpoint 0 automatically

**Reply by paulr34** [reply]
Since it is required we should add this as a high priority item. @brdandu @tecimovic

**Comment by paulr34** [comment]
Matter GUI enhancements:
PR ready for review https://github.com/project-chip/zap/pull/1181

this PR addresses:
https://github.com/project-chip/zap/issues/1145
https://github.com/project-chip/zap/issues/1086
